cdk/resources/containers/coap/src/handler.ts

```diff
@@ -0,0 +1,111 @@
+import { writeFileSync } from 'node:fs'
+import os from 'node:os'
+import path from 'node:path'
+import { execFile } from 'node:child_process'
+import { type Handler } from 'aws-lambda'
+
+type DeviceProperties = {
+	deviceId: string
+	publicKey: string
+	privateKey: string
+	host: string
+	port: number
+}
+
+type Event = { deviceProperties: DeviceProperties; args: string[] }
+
+type EventResponse = {
+	statusCode: number
+	body?: string | string[]
+}
+
+const run = async ({
+	command,
+	args,
+}: {
+	command: string
+	args: string[]
+}): Promise<string[]> => {
+	return new Promise((resolve, reject) => {
+		const cwd = '/function'
+		console.log(`Run command: "${command}" with "${args}" at "${cwd}"`)
+
+		execFile(
+			command,
+			args,
+			{
+				cwd,
+			},
+			(error, stdout, stderr) => {
+				if (error) {
+					return reject(error)
+				}
+
+				const outputs = (stdout || stderr).replace(/^[\d\-:,\s]+|\s*$/gm, '')
+				return resolve(outputs.split(os.EOL))
+			},
+		)
+	})
+}
+
+export const createDeviceProperties = (
+	devicePropertiesLocation: string,
+	deviceProperties: DeviceProperties,
+): void => {
+	const { deviceId, publicKey, privateKey, host, port } = deviceProperties
+	// Generate device.properties
+	const data = `
+deviceId=${deviceId}
+host=${host}
+port=${port}
+privateKey=${privateKey.trim().split('\n').slice(1, -1).join('')}
+publicKey=${publicKey.trim().split('\n').slice(1, -1).join('')}
+    `
```

Comment:
Use [`...`,`...`].join('\n') for better readability.

Reply:
This is desired by CoAP Simulator. They trim out `-----BEGIN CERTIFICATE-----`, `-----END CERTIFICATE-----`, and `\n`

Reply:
But you still can express the same string using array of lines + join, but then can indent the code properly.